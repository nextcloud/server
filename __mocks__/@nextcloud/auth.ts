/**
 * SPDX-FileCopyrightText: 2023 Nextcloud GmbH and Nextcloud contributors
 * SPDX-License-Identifier: AGPL-3.0-or-later
 */
export const getCurrentUser = function() {
	return {
		uid: 'test',
		displayName: 'Test',
		isAdmin: false,
	}
}

export const getRequestToken = function() {
	return 'test-token-1234'
}

export const onRequestTokenUpdate = function() {}
