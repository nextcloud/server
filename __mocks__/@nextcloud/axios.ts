/**
 * SPDX-FileCopyrightText: 2023 Nextcloud GmbH and Nextcloud contributors
 * SPDX-License-Identifier: AGPL-3.0-or-later
 */
export default {
	interceptors: {
		response: {
			use: () => {},
		},
		request: {
			use: () => {},
		},
	},
	get: async () => ({ status: 200, data: {} }),
	delete: async () => ({ status: 200, data: {} }),
	post: async () => ({ status: 200, data: {} }),
}
