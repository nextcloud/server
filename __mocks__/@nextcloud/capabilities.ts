/**
 * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
 * SPDX-License-Identifier: AGPL-3.0-or-later
 */
import type { Capabilities } from '../../apps/files/src/types'

export const getCapabilities = (): Capabilities => {
	return {
		files: {
			bigfilechunking: true,
			blacklisted_files: [],
			forbidden_filename_basenames: [],
			forbidden_filename_characters: [],
			forbidden_filename_extensions: [],
			forbidden_filenames: [],
			undelete: true,
			version_deletion: true,
			version_labeling: true,
			versioning: true,
		},
	}
}
