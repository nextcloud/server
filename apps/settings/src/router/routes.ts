import type { RouteConfig } from 'vue-router'

// Dynamic loading
const AppStore = () => import(/* webpackChunkName: 'settings-apps-view' */'../views/AppStore.vue')
const AppStoreNavigation = () => import(/* webpackChunkName: 'settings-apps-view' */'../views/AppStoreNavigation.vue')
const AppStoreSidebar = () => import(/* webpackChunkName: 'settings-apps-view' */'../views/AppStoreSidebar.vue')

const UserManagement = () => import(/* webpackChunkName: 'settings-users' */'../views/UserManagement.vue')
const UserManagementNavigation = () => import(/* webpackChunkName: 'settings-users' */'../views/UserManagementNavigation.vue')

const routes: RouteConfig[] = [
	{
		name: 'users',
		path: '/:index(index.php/)?settings/users',
		components: {
			default: UserManagement,
			navigation: UserManagementNavigation,
		},
		props: true,
		children: [
			{
				path: ':selectedGroup',
				name: 'group',
			},
		],
	},
	{
		path: '/:index(index.php/)?settings/apps',
		name: 'apps',
		// redirect to our default route - the app discover section
		redirect: {
			name: 'apps-category',
			params: {
				category: 'discover',
			},
		},
		components: {
			default: AppStore,
			navigation: AppStoreNavigation,
			sidebar: AppStoreSidebar,
		},
		children: [
			{
				path: ':category',
				name: 'apps-category',
				children: [
					{
						path: ':id',
						name: 'apps-details',
					},
				],
			},
		],
	},
]

export default routes
