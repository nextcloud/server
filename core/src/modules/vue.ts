import Vue from 'vue'

export * from 'vue'
export default Vue
