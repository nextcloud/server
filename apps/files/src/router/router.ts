/**
 * @copyright Copyright (c) 2022 John Molakvoæ <skjnldsv@protonmail.com>
 *
 * @author John Molakvoæ <skjnldsv@protonmail.com>
 *
 * @license AGPL-3.0-or-later
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 *
 */
import { generateUrl } from '@nextcloud/router'
import queryString from 'query-string'
import Router, { RawLocation, Route } from 'vue-router'
import Vue from 'vue'
import { ErrorHandler } from 'vue-router/types/router'

Vue.use(Router)

// Prevent router from throwing errors when we're already on the page we're trying to go to
const originalPush = Router.prototype.push as (to, onComplete?, onAbort?) => Promise<Route>
Router.prototype.push = function push(to: RawLocation, onComplete?: ((route: Route) => void) | undefined, onAbort?: ErrorHandler | undefined): Promise<Route> {
	if (onComplete || onAbort) return originalPush.call(this, to, onComplete, onAbort)
	return originalPush.call(this, to).catch(err => err)
}

const router = new Router({
	mode: 'history',

	// if index.php is in the url AND we got this far, then it's working:
	// let's keep using index.php in the url
	base: generateUrl('/apps/files'),
	linkActiveClass: 'active',

	routes: [
		{
			path: '/',
			// Pretending we're using the default view
			redirect: { name: 'filelist' },
		},
		{
			path: '/:view/:fileid?',
			name: 'filelist',
			props: true,
		},
	],

	// Custom stringifyQuery to prevent encoding of slashes in the url
	stringifyQuery(query) {
		const result = queryString.stringify(query).replace(/%2F/gmi, '/')
		return result ? ('?' + result) : ''
	},
})

export default router
