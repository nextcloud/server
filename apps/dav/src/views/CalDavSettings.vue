<template>
	<NcSettingsSection :name="$t('dav', 'Calendar server')"
		:doc-url="userSyncCalendarsDocUrl">
		<!-- Can use v-html as:
			- $t passes the translated string through DOMPurify.sanitize,
			- replacement strings are not user-controlled. -->
		<!-- eslint-disable-next-line vue/no-v-html -->
		<p class="settings-hint" v-html="hint" />
		<p>
			<NcCheckboxRadioSwitch id="caldavSendInvitations"
				:checked.sync="sendInvitations"
				type="switch">
				{{ $t('dav', 'Send invitations to attendees') }}
			</NcCheckboxRadioSwitch>
			<!-- Can use v-html as:
				- $t passes the translated string through DOMPurify.sanitize,
				- replacement strings are not user-controlled. -->
			<!-- eslint-disable-next-line vue/no-v-html -->
			<em v-html="sendInvitationsHelpText" />
		</p>
		<p>
			<NcCheckboxRadioSwitch id="caldavGenerateBirthdayCalendar"
				:checked.sync="generateBirthdayCalendar"
				type="switch"
				class="checkbox">
				{{ $t('dav', 'Automatically generate a birthday calendar') }}
			</NcCheckboxRadioSwitch>
			<em>
				{{ $t('dav', 'Birthday calendars will be generated by a background job.') }}
			</em>
			<br>
			<em>
				{{ $t('dav', 'Hence they will not be available immediately after enabling but will show up after some time.') }}
			</em>
		</p>
		<p>
			<NcCheckboxRadioSwitch id="caldavSendEventReminders"
				:checked.sync="sendEventReminders"
				type="switch">
				{{ $t('dav', 'Send notifications for events') }}
			</NcCheckboxRadioSwitch>
			<!-- Can use v-html as:
				- $t passes the translated string through DOMPurify.sanitize,
				- replacement strings are not user-controlled. -->
			<!-- eslint-disable-next-line vue/no-v-html -->
			<em v-html="sendEventRemindersHelpText" />
			<br>
			<em>
				{{ $t('dav', 'Notifications are sent via background jobs, so these must occur often enough.') }}
			</em>
		</p>
		<p class="indented">
			<NcCheckboxRadioSwitch id="caldavSendEventRemindersToSharedGroupMembers"
				:checked.sync="sendEventRemindersToSharedUsers"
				type="switch"
				:disabled="!sendEventReminders">
				{{ $t('dav', 'Send reminder notifications to calendar sharees as well' ) }}
			</NcCheckboxRadioSwitch>
			<em>
				{{ $t('dav', 'Reminders are always sent to organizers and attendees.' ) }}
			</em>
		</p>
		<p class="indented">
			<NcCheckboxRadioSwitch id="caldavSendEventRemindersPush"
				:checked.sync="sendEventRemindersPush"
				type="switch"
				:disabled="!sendEventReminders">
				{{ $t('dav', 'Enable notifications for events via push') }}
			</NcCheckboxRadioSwitch>
		</p>
	</NcSettingsSection>
</template>

<script>
import axios from '@nextcloud/axios'
import { generateUrl } from '@nextcloud/router'
import { loadState } from '@nextcloud/initial-state'
import NcSettingsSection from '@nextcloud/vue/dist/Components/NcSettingsSection.js'
import NcCheckboxRadioSwitch from '@nextcloud/vue/dist/Components/NcCheckboxRadioSwitch.js'

const userSyncCalendarsDocUrl = loadState('dav', 'userSyncCalendarsDocUrl', '#')

export default {
	name: 'CalDavSettings',
	components: {
		NcCheckboxRadioSwitch,
		NcSettingsSection,
	},
	data() {
		return {
			userSyncCalendarsDocUrl,
		}
	},
	computed: {
		hint() {
			const translated = this.$t(
				'dav',
				'Also install the {calendarappstoreopen}Calendar app{linkclose}, or {calendardocopen}connect your desktop & mobile for syncing â†—{linkclose}.',
			)
			return translated
				.replace('{calendarappstoreopen}', '<a target="_blank" href="../apps/office/calendar">')
				.replace('{calendardocopen}', `<a target="_blank" href="${userSyncCalendarsDocUrl}" rel="noreferrer noopener">`)
				.replace(/\{linkclose\}/g, '</a>')
		},
		sendInvitationsHelpText() {
			const translated = this.$t('dav', 'Please make sure to properly set up {emailopen}the email server{linkclose}.')
			return translated
				.replace('{emailopen}', '<a href="../admin#mail_general_settings">')
				.replace('{linkclose}', '</a>')
		},
		sendEventRemindersHelpText() {
			const translated = this.$t('dav', 'Please make sure to properly set up {emailopen}the email server{linkclose}.')
			return translated
				.replace('{emailopen}', '<a href="../admin#mail_general_settings">')
				.replace('{linkclose}', '</a>')
		},
	},
	watch: {
		generateBirthdayCalendar(value) {
			const baseUrl = value ? '/apps/dav/enableBirthdayCalendar' : '/apps/dav/disableBirthdayCalendar'
			axios.post(generateUrl(baseUrl))
		},
		sendInvitations(value) {
			OCP.AppConfig.setValue(
				'dav',
				'sendInvitations',
				value ? 'yes' : 'no'
			)
		},
		sendEventReminders(value) {
			OCP.AppConfig.setValue('dav', 'sendEventReminders', value ? 'yes' : 'no')
		},
		sendEventRemindersToSharedUsers(value) {
			OCP.AppConfig.setValue(
				'dav',
				'sendEventRemindersToSharedUsers',
				value ? 'yes' : 'no'
			)
		},
		sendEventRemindersPush(value) {
			OCP.AppConfig.setValue('dav', 'sendEventRemindersPush', value ? 'yes' : 'no')
		},
	},
}
</script>

<style scoped>
	.indented {
		padding-left: 28px;
	}
	/** Use deep selector to affect v-html */
	* >>> a {
		text-decoration: underline;
	}
	.settings-hint {
		margin-top: -.2em;
		margin-bottom: 1em;
		opacity: .7;
	}
</style>
