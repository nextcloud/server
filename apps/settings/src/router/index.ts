/**
 * SPDX-FileCopyrightText: 2018 Nextcloud GmbH and Nextcloud contributors
 * SPDX-License-Identifier: AGPL-3.0-or-later
 */

import Vue from 'vue'
import Router from 'vue-router'
import { generateUrl } from '@nextcloud/router'
import routes from './routes.ts'

Vue.use(Router)

const router = new Router({
	mode: 'history',
	// if index.php is in the url AND we got this far, then it's working:
	// let's keep using index.php in the url
	base: generateUrl(''),
	linkActiveClass: 'active',
	routes,
})

export default router
