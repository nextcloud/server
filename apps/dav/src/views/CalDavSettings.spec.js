import { render } from '@testing-library/vue'
import CalDavSettings from './CalDavSettings.vue'
// eslint-disable-next-line no-unused-vars
import { generateUrl } from '@nextcloud/router'

jest.mock('@nextcloud/axios')
jest.mock('@nextcloud/router', () => {
	return {
		generateUrl(url) {
			return url
		},
	}
})
jest.mock('@nextcloud/initial-state', () => {
	return {
		loadState: jest.fn(() => 'https://docs.nextcloud.com/server/23/go.php?to=user-sync-calendars'),
	}
})

describe('CalDavSettings', () => {
	const originalOC = global.OC
	const originalOCP = global.OCP

	beforeEach(() => {
		global.OC = { requestToken: 'secret' }
		global.OCP = {
			AppConfig: {
				setValue: jest.fn(),
			},
		}
	})
	afterAll(() => {
		global.OC = originalOC
		global.OCP = originalOCP
	})

	test('interactions', async () => {
		const TLUtils = render(
			CalDavSettings,
			{
				data() {
					return {
						sendInvitations: true,
						generateBirthdayCalendar: true,
						sendEventReminders: true,
						sendEventRemindersToSharedUsers: true,
						sendEventRemindersPush: true,
					}
				},
			},
			Vue => {
				Vue.prototype.$t = jest.fn((app, text) => text)
			}
		)
		expect(TLUtils.container).toMatchSnapshot()
		const sendInvitations = TLUtils.getByLabelText(
			'Send invitations to attendees'
		)
		expect(sendInvitations).toBeChecked()
		const generateBirthdayCalendar = TLUtils.getByLabelText(
			'Automatically generate a birthday calendar'
		)
		expect(generateBirthdayCalendar).toBeChecked()
		const sendEventReminders = TLUtils.getByLabelText(
			'Send notifications for events'
		)
		expect(sendEventReminders).toBeChecked()
		const sendEventRemindersToSharedUsers = TLUtils.getByLabelText(
			'Send reminder notifications to calendar sharees as well'
		)
		expect(sendEventRemindersToSharedUsers).toBeChecked()
		const sendEventRemindersPush = TLUtils.getByLabelText(
			'Enable notifications for events via push'
		)
		expect(sendEventRemindersPush).toBeChecked()

		/*
		FIXME userEvent.click is broken with nextcloud-vue/Button

		await userEvent.click(sendInvitations)
		expect(sendInvitations).not.toBeChecked()
		expect(OCP.AppConfig.setValue).toHaveBeenCalledWith(
			'dav',
			'sendInvitations',
			'no'
		)
		OCP.AppConfig.setValue.mockClear()
		await userEvent.click(sendInvitations)
		expect(sendInvitations).toBeChecked()
		expect(OCP.AppConfig.setValue).toHaveBeenCalledWith(
			'dav',
			'sendInvitations',
			'yes'
		)

		axios.post.mockImplementationOnce((uri) => {
			expect(uri).toBe('/apps/dav/disableBirthdayCalendar')
			return Promise.resolve()
		})
		await userEvent.click(generateBirthdayCalendar)
		axios.post.mockImplementationOnce((uri) => {
			expect(uri).toBe('/apps/dav/enableBirthdayCalendar')
			return Promise.resolve()
		})
		await userEvent.click(generateBirthdayCalendar)
		expect(generateBirthdayCalendar).toBeEnabled()

		OCP.AppConfig.setValue.mockClear()
		await userEvent.click(sendEventReminders)
		expect(sendEventReminders).not.toBeChecked()
		expect(OCP.AppConfig.setValue).toHaveBeenCalledWith(
			'dav',
			'sendEventReminders',
			'no'
		)

		expect(sendEventRemindersToSharedUsers).toBeDisabled()
		expect(sendEventRemindersPush).toBeDisabled()

		OCP.AppConfig.setValue.mockClear()
		await userEvent.click(sendEventReminders)
		expect(sendEventReminders).toBeChecked()
		expect(OCP.AppConfig.setValue).toHaveBeenCalledWith(
			'dav',
			'sendEventReminders',
			'yes'
		)

		expect(sendEventRemindersToSharedUsers).toBeEnabled()
		expect(sendEventRemindersPush).toBeEnabled()
		*/
	})
})
