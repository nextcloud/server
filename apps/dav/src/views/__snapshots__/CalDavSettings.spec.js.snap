// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalDavSettings interactions 1`] = `
<div>
  <div
    class="settings-section settings-section--limit-width"
    data-v-f51cf2d3=""
  >
    <h2
      class="settings-section__name"
      data-v-f51cf2d3=""
    >
       Calendar server 
      <a
        aria-label="External documentation for Calendar server"
        class="settings-section__info"
        data-v-f51cf2d3=""
        href="https://docs.nextcloud.com/server/23/go.php?to=user-sync-calendars"
        rel="noreferrer nofollow"
        target="_blank"
        title="External documentation for Calendar server"
      >
        <span
          aria-hidden="true"
          class="material-design-icon help-circle-icon"
          data-v-f51cf2d3=""
          role="img"
        >
          <svg
            class="material-design-icon__svg"
            fill="currentColor"
            height="20"
            viewBox="0 0 24 24"
            width="20"
          >
            <path
              d="M15.07,11.25L14.17,12.17C13.45,12.89 13,13.5 13,15H11V14.5C11,13.39 11.45,12.39 12.17,11.67L13.41,10.41C13.78,10.05 14,9.55 14,9C14,7.89 13.1,7 12,7A2,2 0 0,0 10,9H8A4,4 0 0,1 12,5A4,4 0 0,1 16,9C16,9.88 15.64,10.67 15.07,11.25M13,19H11V17H13M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12C22,6.47 17.5,2 12,2Z"
            >
              <!---->
            </path>
          </svg>
        </span>
      </a>
    </h2>
    <!---->
    <p
      class="settings-hint"
      data-v-f51cf2d3=""
    >
      Also install the 
      <a
        href="../apps/office/calendar"
        target="_blank"
      >
        Calendar app
      </a>
      , or 
      <a
        href="https://docs.nextcloud.com/server/23/go.php?to=user-sync-calendars"
        rel="noreferrer noopener"
        target="_blank"
      >
        connect your desktop & mobile for syncing â†—
      </a>
      .
    </p>
     
    <p
      data-v-f51cf2d3=""
    >
      <span
        class="checkbox-radio-switch checkbox-radio-switch-switch checkbox-radio-switch--checked"
        data-v-2603be83=""
        data-v-f51cf2d3=""
        style="--icon-size: 36px; --icon-height: 16px;"
      >
        <input
          aria-labelledby="caldavSendInvitations-label"
          class="checkbox-radio-switch__input"
          data-v-2603be83=""
          id="caldavSendInvitations"
          type="checkbox"
          value=""
        />
        <span
          class="checkbox-content checkbox-radio-switch__content checkbox-content-switch checkbox-content--has-text"
          data-v-2603be83=""
          data-v-2672ad1a=""
          id="caldavSendInvitations-label"
        >
          <span
            aria-hidden="true"
            class="checkbox-content__icon checkbox-content__icon--checked checkbox-radio-switch__icon"
            data-v-2672ad1a=""
            inert="inert"
          >
            <span
              aria-hidden="true"
              class="material-design-icon toggle-switch-icon"
              data-v-2672ad1a=""
              role="img"
            >
              <svg
                class="material-design-icon__svg"
                fill="currentColor"
                height="36"
                viewBox="0 0 24 24"
                width="36"
              >
                <path
                  d="M17,7H7A5,5 0 0,0 2,12A5,5 0 0,0 7,17H17A5,5 0 0,0 22,12A5,5 0 0,0 17,7M17,15A3,3 0 0,1 14,12A3,3 0 0,1 17,9A3,3 0 0,1 20,12A3,3 0 0,1 17,15Z"
                >
                  <!---->
                </path>
              </svg>
            </span>
          </span>
          <span
            class="checkbox-content__text checkbox-radio-switch__text"
            data-v-2672ad1a=""
          >
            
			Send invitations to attendees
		
          </span>
        </span>
      </span>
       
      <em
        data-v-f51cf2d3=""
      >
        Please make sure to properly set up 
        <a
          href="../admin#mail_general_settings"
        >
          the email server
        </a>
        .
      </em>
    </p>
     
    <p
      data-v-f51cf2d3=""
    >
      <span
        class="checkbox-radio-switch checkbox checkbox-radio-switch-switch checkbox-radio-switch--checked"
        data-v-2603be83=""
        data-v-f51cf2d3=""
        style="--icon-size: 36px; --icon-height: 16px;"
      >
        <input
          aria-labelledby="caldavGenerateBirthdayCalendar-label"
          class="checkbox-radio-switch__input"
          data-v-2603be83=""
          id="caldavGenerateBirthdayCalendar"
          type="checkbox"
          value=""
        />
        <span
          class="checkbox-content checkbox-radio-switch__content checkbox-content-switch checkbox-content--has-text"
          data-v-2603be83=""
          data-v-2672ad1a=""
          id="caldavGenerateBirthdayCalendar-label"
        >
          <span
            aria-hidden="true"
            class="checkbox-content__icon checkbox-content__icon--checked checkbox-radio-switch__icon"
            data-v-2672ad1a=""
            inert="inert"
          >
            <span
              aria-hidden="true"
              class="material-design-icon toggle-switch-icon"
              data-v-2672ad1a=""
              role="img"
            >
              <svg
                class="material-design-icon__svg"
                fill="currentColor"
                height="36"
                viewBox="0 0 24 24"
                width="36"
              >
                <path
                  d="M17,7H7A5,5 0 0,0 2,12A5,5 0 0,0 7,17H17A5,5 0 0,0 22,12A5,5 0 0,0 17,7M17,15A3,3 0 0,1 14,12A3,3 0 0,1 17,9A3,3 0 0,1 20,12A3,3 0 0,1 17,15Z"
                >
                  <!---->
                </path>
              </svg>
            </span>
          </span>
          <span
            class="checkbox-content__text checkbox-radio-switch__text"
            data-v-2672ad1a=""
          >
            
			Automatically generate a birthday calendar
		
          </span>
        </span>
      </span>
       
      <em
        data-v-f51cf2d3=""
      >
        
			Birthday calendars will be generated by a background job.
		
      </em>
       
      <br
        data-v-f51cf2d3=""
      />
       
      <em
        data-v-f51cf2d3=""
      >
        
			Hence they will not be available immediately after enabling but will show up after some time.
		
      </em>
    </p>
     
    <p
      data-v-f51cf2d3=""
    >
      <span
        class="checkbox-radio-switch checkbox-radio-switch-switch checkbox-radio-switch--checked"
        data-v-2603be83=""
        data-v-f51cf2d3=""
        style="--icon-size: 36px; --icon-height: 16px;"
      >
        <input
          aria-labelledby="caldavSendEventReminders-label"
          class="checkbox-radio-switch__input"
          data-v-2603be83=""
          id="caldavSendEventReminders"
          type="checkbox"
          value=""
        />
        <span
          class="checkbox-content checkbox-radio-switch__content checkbox-content-switch checkbox-content--has-text"
          data-v-2603be83=""
          data-v-2672ad1a=""
          id="caldavSendEventReminders-label"
        >
          <span
            aria-hidden="true"
            class="checkbox-content__icon checkbox-content__icon--checked checkbox-radio-switch__icon"
            data-v-2672ad1a=""
            inert="inert"
          >
            <span
              aria-hidden="true"
              class="material-design-icon toggle-switch-icon"
              data-v-2672ad1a=""
              role="img"
            >
              <svg
                class="material-design-icon__svg"
                fill="currentColor"
                height="36"
                viewBox="0 0 24 24"
                width="36"
              >
                <path
                  d="M17,7H7A5,5 0 0,0 2,12A5,5 0 0,0 7,17H17A5,5 0 0,0 22,12A5,5 0 0,0 17,7M17,15A3,3 0 0,1 14,12A3,3 0 0,1 17,9A3,3 0 0,1 20,12A3,3 0 0,1 17,15Z"
                >
                  <!---->
                </path>
              </svg>
            </span>
          </span>
          <span
            class="checkbox-content__text checkbox-radio-switch__text"
            data-v-2672ad1a=""
          >
            
			Send notifications for events
		
          </span>
        </span>
      </span>
       
      <em
        data-v-f51cf2d3=""
      >
        Please make sure to properly set up 
        <a
          href="../admin#mail_general_settings"
        >
          the email server
        </a>
        .
      </em>
       
      <br
        data-v-f51cf2d3=""
      />
       
      <em
        data-v-f51cf2d3=""
      >
        
			Notifications are sent via background jobs, so these must occur often enough.
		
      </em>
    </p>
     
    <p
      class="indented"
      data-v-f51cf2d3=""
    >
      <span
        class="checkbox-radio-switch checkbox-radio-switch-switch checkbox-radio-switch--checked"
        data-v-2603be83=""
        data-v-f51cf2d3=""
        style="--icon-size: 36px; --icon-height: 16px;"
      >
        <input
          aria-labelledby="caldavSendEventRemindersToSharedGroupMembers-label"
          class="checkbox-radio-switch__input"
          data-v-2603be83=""
          id="caldavSendEventRemindersToSharedGroupMembers"
          type="checkbox"
          value=""
        />
        <span
          class="checkbox-content checkbox-radio-switch__content checkbox-content-switch checkbox-content--has-text"
          data-v-2603be83=""
          data-v-2672ad1a=""
          id="caldavSendEventRemindersToSharedGroupMembers-label"
        >
          <span
            aria-hidden="true"
            class="checkbox-content__icon checkbox-content__icon--checked checkbox-radio-switch__icon"
            data-v-2672ad1a=""
            inert="inert"
          >
            <span
              aria-hidden="true"
              class="material-design-icon toggle-switch-icon"
              data-v-2672ad1a=""
              role="img"
            >
              <svg
                class="material-design-icon__svg"
                fill="currentColor"
                height="36"
                viewBox="0 0 24 24"
                width="36"
              >
                <path
                  d="M17,7H7A5,5 0 0,0 2,12A5,5 0 0,0 7,17H17A5,5 0 0,0 22,12A5,5 0 0,0 17,7M17,15A3,3 0 0,1 14,12A3,3 0 0,1 17,9A3,3 0 0,1 20,12A3,3 0 0,1 17,15Z"
                >
                  <!---->
                </path>
              </svg>
            </span>
          </span>
          <span
            class="checkbox-content__text checkbox-radio-switch__text"
            data-v-2672ad1a=""
          >
            
			Send reminder notifications to calendar sharees as well
		
          </span>
        </span>
      </span>
       
      <em
        data-v-f51cf2d3=""
      >
        
			Reminders are always sent to organizers and attendees.
		
      </em>
    </p>
     
    <p
      class="indented"
      data-v-f51cf2d3=""
    >
      <span
        class="checkbox-radio-switch checkbox-radio-switch-switch checkbox-radio-switch--checked"
        data-v-2603be83=""
        data-v-f51cf2d3=""
        style="--icon-size: 36px; --icon-height: 16px;"
      >
        <input
          aria-labelledby="caldavSendEventRemindersPush-label"
          class="checkbox-radio-switch__input"
          data-v-2603be83=""
          id="caldavSendEventRemindersPush"
          type="checkbox"
          value=""
        />
        <span
          class="checkbox-content checkbox-radio-switch__content checkbox-content-switch checkbox-content--has-text"
          data-v-2603be83=""
          data-v-2672ad1a=""
          id="caldavSendEventRemindersPush-label"
        >
          <span
            aria-hidden="true"
            class="checkbox-content__icon checkbox-content__icon--checked checkbox-radio-switch__icon"
            data-v-2672ad1a=""
            inert="inert"
          >
            <span
              aria-hidden="true"
              class="material-design-icon toggle-switch-icon"
              data-v-2672ad1a=""
              role="img"
            >
              <svg
                class="material-design-icon__svg"
                fill="currentColor"
                height="36"
                viewBox="0 0 24 24"
                width="36"
              >
                <path
                  d="M17,7H7A5,5 0 0,0 2,12A5,5 0 0,0 7,17H17A5,5 0 0,0 22,12A5,5 0 0,0 17,7M17,15A3,3 0 0,1 14,12A3,3 0 0,1 17,9A3,3 0 0,1 20,12A3,3 0 0,1 17,15Z"
                >
                  <!---->
                </path>
              </svg>
            </span>
          </span>
          <span
            class="checkbox-content__text checkbox-radio-switch__text"
            data-v-2672ad1a=""
          >
            
			Enable notifications for events via push
		
          </span>
        </span>
      </span>
    </p>
  </div>
</div>
`;
